### PR TITLE
Parsing config params from v3 links

### PR DESCRIPTION
**Pull request by devinmatte**
## Motivation

Handle existing v3 links, and redirect them to the equivalent v4 page

https://github.com/transitmatters/t-performance-dash/issues/428
https://github.com/transitmatters/t-performance-dash/issues/414

## Changes

- Adds hook to collect config param, parse it, and redirect to equivalent v4 route

Currently handling `rapidtransit` and `bus`, not handling any other routes

### CODE REVIEW

File: common/components/widgets/Widget.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ interface WidgetPageProps {
 
 export const WidgetPage: React.FC<WidgetPageProps> = ({ children }) => {
   return (
-    <div className={classNames('flex w-full flex-1 flex-col gap-y-2 pb-28 lg:p-0')}>
```
[the remark this answers is not in the document]
Comment (devinmatte) [reply common/components/widgets/Widget.tsx:11]: Yes that's why I did this. It looks much better on mobile now

File: pages/bus.tsx
[the remark this answers is not in the document]
Comment (devinmatte) [reply pages/bus.tsx:1]: Yeah middleware probably works better. If we'd rather go that route I can give this another shot with that instead

**Comment by devinmatte** [comment]
Using the middleware now. It works much better too!

**Review by PatrickCleary — COMMENTED** [review]
Works great! Thanks for putting this together. Definitely gonna be important for v3 -> v4 transition.

It also made me realize we should have date values auto-fill when not present in the URL, like how stations autofill. I'll work on that.

One thing I noticed in testing is that for single date URL conversion, the new setup needs a `queryType=single` value to function properly. I know you're not in agreement that is the way to go, so if you want to leave it for now that's fine with me.

But I think there's 2 ways to potentially approach it. Either add the queryType in the middleware when only one date is present, or have logic in the app that adds the `queryType` automatically whenever we're in the `trips` section and the value is not present. Lmk what you think, or if you wanna skip this for now.

I opened this issue #497 which is preexisting but I noticed it while testing.

File: middleware.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        };
+      }
+    }
+  }
+};
+
+// This function can be marked `async` if using `await` inside
+export function middleware(request: NextRequest) {
+  const { search } = request.nextUrl;
+  const resultParams = configToQueryParams(search);
```
[the remark this answers is not in the document]
Comment (devinmatte) [reply middleware.ts:52]: That's covered by line 69. It only fires for `/bus` not for `/bus/trips` for example

**Review by skaplan-dev — APPROVED** [review]

**Pull request merged**